code/modules/mob/living/carbon/human/life.dm

```diff
@@ -742,178 +782,166 @@
-		if(!machine.check_eye(src))
-			reset_perspective(null)
-	else
-		var/isRemoteObserve = 0
-		if((REMOTE_VIEW in mutations) && remoteview_target)
-			isRemoteObserve = 1
-
-			if(remoteview_target.stat != CONSCIOUS)
-				to_chat(src, "<span class='alert'>Your psy-connection grows too faint to maintain!</span>")
-				isRemoteObserve = 0
-
-			if(PSY_RESIST in remoteview_target.mutations)
-				to_chat(src, "<span class='alert'>Your mind is shut out!</span>")
-				isRemoteObserve = 0
-
-			// Not on the station or mining?
-			var/turf/temp_turf = get_turf(remoteview_target)
-			if(!(temp_turf in config.contact_levels))
-				to_chat(src, "<span class='alert'>Your psy-connection grows too faint to maintain!</span>")
-				isRemoteObserve = 0
-
-		if(remote_view)
-			isRemoteObserve = 1
-
-		if(!isRemoteObserve && client && !client.adminobs)
-			remoteview_target = null
-			reset_perspective(null)
-
-/mob/living/carbon/human/handle_hud_icons()
-	dna.species.handle_hud_icons(src)
-
-/mob/living/carbon/human/handle_hud_icons_health()
-	dna.species.handle_hud_icons_health(src)
-	handle_hud_icons_health_overlay()
+			switch(hal_screwyhud)
+				if(SCREWYHUD_CRIT)
+					healths.icon_state = "health6"
+				if(SCREWYHUD_DEAD)
+					healths.icon_state = "health7"
+				if(SCREWYHUD_HEALTHY)
+					healths.icon_state = "health0"
+				else
+					switch(get_perceived_trauma())
+						if(100 to INFINITY)
+							healths.icon_state = "health0"
+						if(80 to 100)
+							healths.icon_state = "health1"
+						if(60 to 80)
+							healths.icon_state = "health2"
+						if(40 to 60)
+							healths.icon_state = "health3"
+						if(20 to 40)
+							healths.icon_state = "health4"
+						if(0 to 20)
+							healths.icon_state = "health5"
+						else
+							healths.icon_state = "health6"
+
+	if(healthdoll)
+		if(stat == DEAD || (status_flags & FAKEDEATH))
+			healthdoll.icon_state = "healthdoll_DEAD"
+			if(healthdoll.overlays.len)
+				healthdoll.overlays.Cut()
+		else
+			var/list/new_overlays = list()
+			var/list/cached_overlays = healthdoll.cached_healthdoll_overlays
+			// Use the dead health doll as the base, since we have proper "healthy" overlays now
+			healthdoll.icon_state = "healthdoll_DEAD"
+			for(var/obj/item/organ/external/O in bodyparts)
+				var/damage = O.burn_dam + O.brute_dam
+				var/comparison = (O.max_damage/5)
+				var/icon_num = 0
+				if(damage)
+					icon_num = 1
+				if(damage > (comparison))
+					icon_num = 2
+				if(damage > (comparison*2))
+					icon_num = 3
+				if(damage > (comparison*3))
+					icon_num = 4
+				if(damage > (comparison*4))
+					icon_num = 5
+				new_overlays += "[O.limb_name][icon_num]"
+			healthdoll.overlays += (new_overlays - cached_overlays)
+			healthdoll.overlays -= (cached_overlays - new_overlays)
+			healthdoll.cached_healthdoll_overlays = new_overlays
+
+/mob/living/carbon/human/proc/handle_nutrition_alerts() //This is a terrible abuse of the alert system; something like this should be a HUD element
```

Comment:
This needs to be made into a screen element---this is currently abusing the alert system.